routes/index.js

```diff
@@ -1,142 +1,166 @@
 const game = new (require("../game/game"))();
 const Password = require("../game/password");
 
-router.get("/", (req, res) => {
-    res.render("index", {
-        title: "HealtyBoiGame",
-        message: "Hello there!",
-        loggedInId: req.session.accountId,
-        accounts: game.accounts,
-    });
-});
+const { generateToken } = require("../utils/jwtUtils");
 
 router.post("/", (req, res) => {
-    const account = game.getAccountById(req.session.accountId);
+    let modalInfo = null;
+    const account = game.getAccountById(req.user.id);
     if (!account) {
-        invalidAccountRender(res, req.session.accountId);
+        res.status(401).json({
+            success: false,
+            message: "Could not find any account tied to your token",
+        });
         return;
     }
-    let modalInfo = null;
-    let xpMultiplier = 1;
-    if ("remove-xp" in req.body) {
-        xpMultiplier = -1;
-    }
-    const amount = 10 * xpMultiplier;
-    const leveledUp = account.addXp(amount);
-
-    if (leveledUp) {
-        modalInfo = {
-            title: "You Leveled Up!",
-            description: `You leveled up to level ${account.level}`,
-            extraHtml: '<i class="nes-icon is-large star"></i>',
-        };
+
+    let target = null;
+    switch (req.body.action) {
+        case "attack":
+            target = game.getAccountById(req.body.id);
+            target.damage();
+            break;
+        case "heal":
+            target = game.getAccountById(req.body.id);
+            target.heal();
+            break;
+        default:
+            let xpMultiplier = 1;
+            if ("remove-xp" in req.body) {
+                xpMultiplier = -1;
+            }
+            const amount = 10 * xpMultiplier;
+            const leveledUp = account.addXp(amount);
+
+            if (leveledUp) {
+                modalInfo = {
+                    title: "You Leveled Up!",
+                    description: `You leveled up to level ${account.level}`,
+                    extraHtml: '<i class="nes-icon is-large star"></i>',
+                };
+            }
     }
 
-    res.render("index", {
-        title: "HealtyBoiGame",
-        message: "Hello there!",
-        accounts: game.accounts,
-        loggedInId: req.session.accountId,
+    res.json({
+        success: true,
+        accounts: game.accounts.map((a) => ({
+            ...a,
+            passwordHash: "",
+            passwordSalt: "",
+        })),
         modalInfo,
     });
 });
 
 router.get("/account", (req, res) => {
-    res.send(
-        JSON.stringify(
-            game.accounts.map((a) => {
-                const obj = { ...a };
-                obj.passwordHash = "";
-                obj.passwordSalt = "";
-                return obj;
-            }),
-        ),
+    res.json(
+        game.accounts.map((a) => {
+            const obj = { ...a };
+            obj.passwordHash = "";
+            obj.passwordSalt = "";
+            return obj;
+        }),
```

Comment:
this thing you did was nicer

```suggestion
        game.accounts.map((a) => ({
            ...a,
            passwordHash: "",
            passwordSalt: "",
        })),
```